### PR TITLE
Vendor operatorkit updates

### PR DESCRIPTION
**Pull request by MarcelMue**
I tried to make operatorkit happy while not breaking too much in this PR. Not sure if this is the right way to do things, but I saw this beign done similarly in other operators.

towards #12

### CODE REVIEW

File: service/resource/endpoint/create.go [hunk 1]
```diff
@@ -9,15 +9,15 @@ import (
 	apiv1 "k8s.io/client-go/pkg/api/v1"
 )
 
-func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (Endpoint, error) {
```
Comment (xh3b4sd) [inline service/resource/endpoint/create.go]: The return value should remain `interface{}`.

File: service/resource/endpoint/create.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desire
 	return createState, nil
 }
 
-func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState interface{}) error {
+func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createState interface{}) error {
```
Comment (xh3b4sd) [inline service/resource/endpoint/create.go]: Lets please sort the methods alphabetically, while public methods are at the top and private ones at the bottom.

File: service/resource/endpoint/create_test.go [hunk 3]
```diff
@@ -148,7 +148,7 @@ func Test_Resource_Endpoint_GetCreateState(t *testing.T) {
 		}
 	}
 	for i, tc := range testCases {
-		result, err := newResource.GetCreateState(context.TODO(), tc.Obj, tc.CurrentState, tc.DesiredState)
+		result, err := newResource.newCreateChange(context.TODO(), tc.Obj, tc.CurrentState, tc.DesiredState)
```
Comment (xh3b4sd) [inline service/resource/endpoint/create_test.go]: The test function name should be renamed accordingly.

File: service/resource/endpoint/create.go (hunk 1 above)
Comment (MarcelMue) [reply service/resource/endpoint/create.go]: Done, I saw this in `azure-operator` I believe, so I wasn't sure.
Here to be exact: https://github.com/giantswarm/azure-operator/blob/cc829fc9665e8ef51ebe393061cfca8f56625a09/service/resource/deployment/resource.go#L269
So is this a preference or should we be consistent with this?

File: service/resource/endpoint/create.go (hunk 2 above)
Comment (MarcelMue) [reply service/resource/endpoint/create.go]: Fixed.

File: service/resource/endpoint/create_test.go (hunk 3 above)
Comment (MarcelMue) [reply service/resource/endpoint/create_test.go]: Renamed and reordered.

File: service/resource/endpoint/create.go [hunk 4]
```diff
@@ -91,3 +62,32 @@ func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState inte
 
 	return nil
 }
+
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
```
Comment (xh3b4sd) [inline service/resource/endpoint/create.go]: @kopiczko do you think the private `new*Change` methods should return a specific type or should they also simply return `interface{}`. I Migrated so far by keeing `interface{}`. We just saw that you in azure-operator used `[]Deployment` as specific type. How should we do this and why?

File: service/resource/endpoint/create.go (hunk 1 above)
Comment (xh3b4sd) [reply service/resource/endpoint/create.go]: Interesting. I have missed this in the azure-operator review. I asked Pawel. Lets see what we can come up with. Shouldn't block us now.

File: service/resource/endpoint/create.go (hunk 4 above)
Comment (kopiczko) [reply service/resource/endpoint/create.go]: I prefer using a proper type whenever we can.  You can switch that from `[]Deployment` to `interface{}` and this still compiles and works the same. Just a little bit more type safety.

I find this especially beneficial when `New*Patch` is dump. It protects against returning nil when the `new*Change` is expected to return a non-{pointer,struct,map} value.

But this is interchangeable and no real benefit besides type safety.

> I Migrated so far by keeing interface{}.

I see no issue with that. I think I did the same in cert-oeprator. Let's take a common approach in new Resources though. I vote for proper type.

Comment (xh3b4sd) [reply service/resource/endpoint/create.go]: > I vote for proper type.

Agreed.

**Pull request merged**